rewrite-groovy/src/main/java/org/openrewrite/groovy/GroovyParserVisitor.java

```diff
@@ -2522,15 +2529,42 @@ private int sourceLengthOfString(ConstantExpression expr) {
         return lengthAccordingToAst;
     }
 
-    private static @Nullable Integer getInsideParenthesesLevel(ASTNode node) {
+    private @Nullable Integer getInsideParenthesesLevel(ASTNode node) {
         Object rawIpl = node.getNodeMetaData("_INSIDE_PARENTHESES_LEVEL");
         if (rawIpl instanceof AtomicInteger) {
             // On Java 11 and newer _INSIDE_PARENTHESES_LEVEL is an AtomicInteger
             return ((AtomicInteger) rawIpl).get();
-        } else {
+        } else if (rawIpl instanceof Integer) {
             // On Java 8 _INSIDE_PARENTHESES_LEVEL is a regular Integer
             return (Integer) rawIpl;
+        } else if (node instanceof MethodCallExpression) {
+            MethodCallExpression expr = (MethodCallExpression) node;
+            int saveCursor = cursor;
+            whitespace();
+            int count = determineParenthesisLevel(expr.getObjectExpression().getLineNumber(), expr.getLineNumber(), expr.getObjectExpression().getColumnNumber(), expr.getColumnNumber());
+            cursor = saveCursor;
+            return count;
+        }
+        return null;
+    }
+
+    private int determineParenthesisLevel(int childLineNumber, int parentLineNumber, int childColumn, int parentColumn) {
```

Comment:
To be honest, I don't think this abstraction is right, because the cursor changes are implied from outside of this function. With other words, this function does only work well if you use it exactly like it is meant to be.

I would go the `MethodCallExpression` as argument. That way it's more in line with the `getInsideParenthesesLevel(ASTNode node` (just an overloaded function with more explicit Type) and it's also works from everywhere because the cursor travel in all kept within this function. But for starters, I would include the `saveCursor` part within this function. Changing the argument to `MethodCallExpression` is up to you!

Reply:
I moved the `saveCursor` and related `whitespace` parts and added JavaDoc to the method. This setup makes it easier to reuse when another case is found where the AST falls short